src/main/java/king/King.java

```diff
@@ -0,0 +1,302 @@
+        String taskEndTime = "";
+        boolean existFrom = false;
+        boolean existTo = false;
+
+        try {
+            for (int i = 0; i < taskSpecifics.length; i++) {
+                if (taskSpecifics[i].equals("/from")) {
+                    seperationIndexFirst = i;
+                    existFrom = true;
+
+                } else if (taskSpecifics[i].equals("/to")) {
+                    seperationIndexSecond = i;
+                    existTo = true;
+                }
+            }
+
+            if (!existTo && !existFrom) {
+                throw new KingException("Both starting & ending time not detected. Please try again:)\n");
+            }
+            if (!existFrom) {
+                throw new KingException("Starting time not detected. Please try again:)\n");
+            } else if (!existTo) {
+                throw new KingException("Ending time not detected. Please try again:)\n");
+            }
+
+
+            for (int i = 1; i < seperationIndexFirst; i++) {
+                taskContent += taskSpecifics[i];
+                if (i < seperationIndexFirst - 1) {
+                    taskContent += " ";
+                }
+            }
+            for (int i = seperationIndexFirst + 1; i < seperationIndexSecond; i++) {
+                taskStartTime += taskSpecifics[i];
+                if (i < seperationIndexSecond - 1) {
+                    taskStartTime += " ";
+                }
+            }
+            for (int i = seperationIndexSecond + 1; i < taskSpecifics.length; i++) {
+                taskEndTime += taskSpecifics[i];
+                if (i < taskSpecifics.length - 1) {
+                    taskEndTime += " ";
+                }
+            }
+        } catch (KingException e) {
+            System.out.println(LINE + e.getMessage() + LINE);
+            return;
+        }
+
+        Task t = new Event(taskContent, taskStartTime, taskEndTime);
+        tasks.add(t);
+        tasksCount += 1;
+        printAddedTaskDescription(t);
+
+        FileAccess.updateFile();
+    }
+
+    private static void printAddedTaskDescription(Task t) {
+        System.out.println(LINE + "Got it. I've added this task:\n   "
+                           + t.getTaskDescription() +
+                           "\nNow you have " + tasksCount + " tasks in the list.\n" + LINE);
+    }
+
+    private static void handleUserInputExceptions(String userInput) throws KingException {
+        userInput = userInput.toLowerCase();
+
+        if (userInput.isEmpty()) {
+            throw new KingException("You have empty input! How can I help you?\n");
+        } else if (!((userInput.startsWith("todo")) || userInput.startsWith("deadline")
+                     || userInput.startsWith("event") || userInput.startsWith("list")
+                     || userInput.startsWith("mark") || userInput.startsWith("unmark")
+                     || userInput.startsWith("bye") || userInput.startsWith("delete"))) {
```

Comment:
Maybe you could refactor this complicated expression or extract it to a boolean variable